crates/svm-sdk/src/host/ext.rs

```diff
@@ -0,0 +1,252 @@
+use crate::host::Host;
+use crate::memory;
+use crate::{Address, Amount, LayerId};
+
+extern crate alloc;
+extern crate std;
+
+use alloc::string::String;
+
+use std::sync::{Mutex, MutexGuard};
+use std::vec::Vec;
+
+/// ### `offset` meaning in this file:
+///
+/// The parameter `offset here denotes a memory address integer serving as a pointer to a cell to the current
+/// running WASM instance. Counting is zero-based.
+
+/// ## SVM Imports  
+///
+/// WASM Imports under namespace `svm` for SVM programs.
+/// Each running SVM app can assume their existence regardless of
+/// the additional imports given by the host (a.k.a `Node / Full-Node`).
+#[link(wasm_import_module = "svm")]
+extern "C" {
+    /// Returns the memory offset where the transaction's input `calldata` starts.
+    fn svm_calldata_offset() -> u32;
+
+    /// Returns the transaction's input `calldata` byte-length.
+    fn svm_calldata_len() -> u32;
+
+    /// Signals to SVM that the current running transaction output (a.k.a `returndata`)
+    /// lays out in memory starting from offset `offset` and its byte-length is `length`.
+    fn svm_returndata(offset: u32, length: u32);
```

Comment:
Calling it `svm_set_returndata`, same as in the `Host` trait, will make more sense.

Reply:
np (I thought it'll be nicer to keep it short but not critical)